clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -140,9 +144,29 @@ public boolean mkdirs(Path path, FsPermission fsPermission) throws IOException {
 
     @Override
     public FileStatus getFileStatus(Path path) throws IOException {
-        LOG.debug("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ getFileStatus, path: {} $$$$$$$$$$$$$$$$$$$$$$$$$$$$", path.toString());
-        FileStatus fStatus = new FileStatus(0, false, 1, 20, 1, path);
-        return fStatus;
+        Location loc = pathToLocation(path);
+        if (loc == null) {
+            throw new FileNotFoundException(path.toString());
+        }
+        try {
+            ObjectsApi objectsApi = new ObjectsApi(this.apiClient);
+            ObjectStats objectStat = objectsApi.statObject(loc.getRepository(), loc.getRef(), loc.getPath());
+            long length = 0;
+            Long sizeBytes = objectStat.getSizeBytes();
+            if (sizeBytes != null) {
+                length = sizeBytes;
+            }
+            long modificationTime = 0;
+            Long mtime = objectStat.getMtime();
+            if (mtime != null) {
+                modificationTime = mtime * 1000; // convert to ms
```

Comment:
you can use TimeUnit.tomillis here (e.g. https://www.geeksforgeeks.org/timeunit-tomillis-method-in-java-with-examples/)

Reply:
cool